crates/bitwarden-vault/src/totp.rs

```diff
@@ -166,9 +168,28 @@ impl FromStr for Totp {
 
         let params = if key.starts_with("otpauth://") {
             let url = Url::parse(&key).map_err(|_| TotpError::InvalidOtpauth)?;
+            let label = remove_first(url.path());
+            let label_string = label.map(|v| v.to_string());
+            let pieces: Vec<String> = label
+                .unwrap_or("")
+                .split(":")
+                .map(|v| v.to_string())
+                .collect();
+            let label_issuer = if pieces.len() > 1 {
+                pieces.first()
+            } else {
+                None
+            };
+            let label_account = if pieces.len() > 1 {
+                pieces.get(1)
+            } else {
+                label_string.as_ref()
+            };
```

Comment:
There's a bit to unpack here. We generally don't need to take ownership of strings unless we use them as a return value. Meaning it's not until line 192 and 210 that we really have to call `clone`, or `to_string` on the inner value of the option, i.e. `opt.map(|v| v.to_string())`.

Removing those simplifies the code a bit.

```suggestion
          let pieces: Vec<&str> = label.unwrap_or("").split(":").collect();
            let label_issuer = if pieces.len() > 1 {
                pieces.first()
            } else {
                None
            };
            let label_account = if pieces.len() > 1 {
                pieces.get(1)
            } else {
                label.as_ref()
            };
```